jtelegrambotapi-core/src/main/java/com/jtelegram/api/commands/CommandHandler.java

```diff
@@ -1,7 +1,20 @@
 package com.jtelegram.api.commands;
 
-public interface CommandHandler {
+import com.jtelegram.api.commands.filters.CommandFilter;
+import com.jtelegram.api.events.message.TextMessageEvent;
 
-    void onCommand(Command command);
+public abstract class CommandHandler extends CommandFilter {
+
+    public abstract void onCommand(TextMessageEvent event, Command command);
+
+    @Override
+    protected final boolean _test(TextMessageEvent event, Command command) {
```

Comment:
Let's not use methods with underlines. AFAIK that is out of the java conventions.

Reply:
```
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter
of each internal word capitalized.
```

[Code Conventions for the Java Programming Language: 9. Naming Conventions](http://www.oracle.com/technetwork/java/codeconventions-135099.html)